pkg/rules/handlers.go

```diff
@@ -0,0 +1,267 @@
+package rules
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/http/types"
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+	"github.com/go-chi/chi/v5"
+)
+
+type API struct {
+	service ServiceInterface
+
+	logger logging.LoggerInterface
+}
+
+func (a *API) RegisterEndpoints(mux *chi.Mux) {
+	mux.Get("/api/v0/rules", a.handleList)
+	mux.Get("/api/v0/rules/{id}", a.handleDetail)
+	mux.Post("/api/v0/rules/{id}", a.handleCreate)
+	mux.Put("/api/v0/rules/{id}", a.handleUpdate)
+	mux.Delete("/api/v0/rules/{id}", a.handleRemove)
+}
+
+func (a *API) handleList(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	limit, err := strconv.ParseInt(r.URL.Query().Get("limit"), 10, 64)
+
+	if err != nil {
+		limit = 100
+	}
+
+	offset, err := strconv.ParseInt(r.URL.Query().Get("offset"), 10, 64)
+
+	if err != nil {
+		offset = 0
+	}
+
+	rules, err := a.service.ListRules(r.Context(), limit, offset)
+
+	if err != nil {
+
+		rr := types.Response{
+			Status:  http.StatusInternalServerError,
+			Message: err.Error(),
+		}
+
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(rr)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(
+		types.Response{
+			Data:    rules,
+			Message: "List of rules",
+			Status:  http.StatusOK,
+		},
+	)
+}
+
+func (a *API) handleDetail(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	ruleID := chi.URLParam(r, "id")
+	ruleID = cleanStr(ruleID)
+
+	rule, err := a.service.GetRule(r.Context(), ruleID)
+
+	if err != nil {
+
+		rr := types.Response{
+			Status:  http.StatusInternalServerError,
+			Message: err.Error(),
+		}
+
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(rr)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(
+		types.Response{
+			Data:    rule,
+			Message: "Rule detail",
+			Status:  http.StatusOK,
+		},
+	)
+}
+
+func (a *API) handleCreate(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	ruleID := chi.URLParam(r, "id")
+	ruleID = cleanStr(ruleID)
+	fileName := r.URL.Query().Get("filename")
```

Comment:
this needs to come from the payload somehow

Reply:
sure

Reply:
The filename is now generated from the id.